src/dpp/presence.cpp

```diff
@@ -135,18 +183,56 @@ presence& presence::fill_from_json(nlohmann::json* j) {
 		for (auto & act : (*j)["activities"]) {
 			activity a;
 			a.name = string_not_null(&act, "name");
+			a.details = string_not_null(&act, "details");
+			if (act.find("assets") != act.end()) {
+				a.assets.large_image = string_not_null(&(act["assets"]), "large_image");
+				a.assets.large_text = string_not_null(&(act["assets"]), "large_text");
+				a.assets.small_image = string_not_null(&(act["assets"]), "small_image");
+				a.assets.small_text = string_not_null(&(act["assets"]), "small_text");
+			}
 			a.state = string_not_null(&act, "state"); // if user
-			if (a.state.empty()) a.state = string_not_null(&act, "details"); // if activity from bot, maybe?
```

Comment:
why is this going away?

Reply:
Because i created a new variable vor that.
stats and details can and will be send together so people want them both